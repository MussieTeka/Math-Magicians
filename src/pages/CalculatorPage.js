import React from 'react';
import Calculator from '../components/Calculator';
import '../styles/pages/CalculatorPage.css';

function CalculatorPage() {
  return (
    <div className="calculator-page-container">
      <div className="calculator-description">
        <h1>Let&apos;s do some Maths!</h1>
        <p>
          Use this calculator to perform basic arithmetic operations such as
          addition, subtraction, multiplication, and division.
        </p>
      </div>
      <div className="calculator-container">
        <Calculator />
      </div>
    </div>
  );
}

export default CalculatorPage;
