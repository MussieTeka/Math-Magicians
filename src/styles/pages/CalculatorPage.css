.calculator-page-container {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  padding: 40px 0;
  height: 100%;
  width: 100%;
}

.calculator-description {
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  padding: 20px;
}

.calculator-container {
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
}

.calculator-description h1 {
  font-family: 'Montserrat', sans-serif;
  font-size: 36px;
  font-weight: bold;
  margin-bottom: 20px;
  color: #4d4d4d;
}

.calculator-description p {
  font-family: 'Open Sans', sans-serif;
  font-size: 18px;
  line-height: 1.6;
  margin-bottom: 40px;
  color: #666;
}

@media (max-width: 768px) {
  .calculator-page-container {
    flex-direction: column;
    padding: 20px;
  }

  .calculator-description {
    padding: 0;
    margin-bottom: 20px;
    text-align: center;
    align-items: center;
  }

  .calculator-container {
    width: 100%;
  }

  .calculator-description h1 {
    font-size: 24px;
    margin-bottom: 10px;
  }

  .calculator-description p {
    font-size: 16px;
    margin-bottom: 20px;
  }
}
