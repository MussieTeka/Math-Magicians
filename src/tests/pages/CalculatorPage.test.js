import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorPage from '../../pages/CalculatorPage';

describe('CalculatorPage component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CalculatorPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
