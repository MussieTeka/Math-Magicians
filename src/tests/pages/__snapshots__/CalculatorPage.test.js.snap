// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorPage component renders correctly 1`] = `
<div
  className="calculator-page-container"
>
  <div
    className="calculator-description"
  >
    <h1>
      Let's do some Maths!
    </h1>
    <p>
      Use this calculator to perform basic arithmetic operations such as addition, subtraction, multiplication, and division.
    </p>
  </div>
  <div
    className="calculator-container"
  >
    <div
      className="calculator"
    >
      <div
        className="display"
      >
        0
      </div>
      <div
        className="buttons"
      >
        <button
          className="ac"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="pm"
          onClick={[Function]}
          type="button"
        >
          ±
        </button>
        <button
          className="percent"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          ×
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          −
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="zero number"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="number"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="equal operator"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
